crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -0,0 +1,189 @@
+    let native_object: NativeSuiObject = bcs::from_bytes(serialized_object)
+        .map_err(|e| Error::Internal(format!("Failed to deserialize object: {e}")).extend())?;
+    let move_object = native_object.data.try_as_move().ok_or_else(|| {
+        Error::Internal("Failed to convert object into MoveObject".to_string()).extend()
+    })?;
+    let struct_tag = native_object
+        .data
+        .struct_tag()
+        .ok_or_else(|| Error::Internal("StructTag missing on object".to_string()).extend())?;    
+    let contents = move_object.contents();
+    let type_tag = TypeTag::Struct(Box::new(struct_tag.clone()));    
+    let move_type_layout = cache.type_layout(type_tag.clone()).await?;
+    let move_struct = match move_type_layout {
+        MoveTypeLayout::Struct(move_struct_layout) => {
+            MoveStruct::simple_deserialize(contents, &move_struct_layout)
+        }
+        _ => Err(Error::Internal("Object is not a move struct".to_string()).extend())?,
+    }?;
+    Ok((struct_tag, move_struct))
+}
+
+pub fn extract_field_from_move_struct(
+    move_struct: MoveStruct,
+    field_name: &str,
+) -> Result<value::MoveValue> {
+    match move_struct {
+        MoveStruct::WithTypes { fields, .. } => {
+            fields.into_iter().find_map(|(id, value)| {
+                if id.to_string() == field_name {
+                    Some(value)
+                } else {
+                    None
+                }
+            })
+        }
+        .ok_or_else(|| Error::Internal(format!("Field '{}' not found", field_name)).extend()),
+        _ => Err(Error::Internal("Unexpected Move struct type".to_string()).extend()),
+    }
+}
+
+#[Object]
+impl DynamicField {
+    async fn name(&self, ctx: &Context<'_>) -> Result<Option<MoveValue>> {
+        let cache: &PackageCache = ctx.data().map_err(|_| {
+            graphql_error(
+                code::INTERNAL_SERVER_ERROR,
+                "Unable to fetch Package Cache.",
+            )
+        })?;
+        let (struct_tag, move_struct) =
+            deserialize_move_struct(&self.stored_object.serialized_object, cache)
+                .await
+                .extend()?;
+
+        // Get TypeTag of the DynamicField name from StructTag of the MoveStruct
+        let dynamic_field_type = self.dynamic_field_type()?;
+        let type_tag = DynamicFieldInfo::try_extract_field_name(&struct_tag, &dynamic_field_type)
+            .map_err(|e| Error::Internal(e.to_string()).extend())?;
+        let undecorated: Option<value::MoveValue>;
+
+        let name_move_value = extract_field_from_move_struct(move_struct, "name")?;
+
+        if self.is_dynamic_object() {
+            let inner_name_move_value = match name_move_value {
+                value::MoveValue::Struct(inner_struct) => {
+                    extract_field_from_move_struct(inner_struct, "name")
+                }
+                _ => Err(Error::Internal("Unexpected wrapper type".to_string()).extend()),
```

Comment:
Message seems wrong here -- you do expect a wrapper type in this case, and you didn't get one.